File: billing/app/dto/BillDTO.py
from tkinter import W
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel

from .BillPartDTO import BillPartDTO


class BillDTO(BaseModel):
    id: Optional[UUID] = None
    userId: UUID
    eventId: UUID
    bucketName: Optional[str] = None
    path: Optional[str] = None
    parts: Optional[List[BillPartDTO]] = None
    total_price: Optional[float] = None

    model_config = {"from_attributes": True}

File: frontend/src/types/BillDTO.ts
export interface BillPartDTO {
  id: string | null
  billId: string
  userId: string | null
  price: number
  text: string
}

export interface BillDTO {
  id: string | null
  userId: string
  eventId: string
  bucketName: string | null
  path: string
  total_price: number
  parts: BillPartDTO[] | null | []
}
